pipe.go

```diff
@@ -108,12 +20,31 @@ func (p *PipeReader) Read(data []byte) (int, error) {
 
 // Close unregisters the reader from the logger.
 func (p *PipeReader) Close() error {
-	p.sink.RemoveReader(p)
+	if p.core != nil {
+		loggerCore.DeleteCore(p.core)
+	}
 	return p.r.Close()
 }
 
-// NewPipeReader creates a new in-memory reader that reads from the logger.
+// NewPipeReader creates a new in-memory reader that reads from all loggers
+// using the default format.
 // The caller must call Close on the returned reader when done.
 func NewPipeReader() *PipeReader {
-	return pipes.NewReader()
+	return NewPipeReaderFormat(primaryFormat)
+}
+
+// NewPipeReader creates a new in-memory reader that reads from all loggers
+// using the supplied format.
+// The caller must call Close on the returned reader when done.
+func NewPipeReaderFormat(format LogFormat) *PipeReader {
```

Comment:
nit: I'd consider using the functional option pattern here instead of introducing a new constructor.

* We haven't cut a release yet so it won't be a breaking change.
* I'd love to be able to set the level as well.

https://dave.cheney.net/2014/10/17/functional-options-for-friendly-apis

Reply:
Done. Added `PipeFormat` and `PipeLevel` option functions.